src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -2498,6 +2499,12 @@ instance ToAbstract C.Pragma where
             "INJECTIVE used on ambiguous name " ++ prettyShow x
           _        -> genericError "Target of INJECTIVE pragma should be a defined symbol"
       return [ A.InjectivePragma y ]
+  toAbstract (C.InjectiveForInferencePragma _ x) = do
+      e <- toAbstract $ OldQName x Nothing
+      y <- case e of
+          A.Def  x -> return x
+          _        -> genericError "Target of INJECTIVE_FOR_INFERENCE pragma should be a defined symbol"
```

Comment:
Could we have a named error (rather than `genericError`)?

Reply:
The answer is yes, but almost all errors in that function are `genericError`.